pkg/adapter/github/adapter.go

```diff
@@ -73,135 +79,192 @@ func (a *Adapter) handleEvent(payload interface{}, hdr http.Header) error {
 
 	log.Printf("Handling %s", gitHubEventType)
 
-	cloudEventType := sourcesv1alpha1.GetGitHubSourceEventType(gitHubEventType)
-	source, err := sourceFromGitHubEvent(gh.Event(gitHubEventType), payload)
-	if err != nil {
-		return err
-	}
+	cloudEventType := sourcesv1alpha1.GitHubEventType(gitHubEventType)
+	subject := subjectFromGitHubEvent(gh.Event(gitHubEventType), payload)
+
+	eventContext := cloudevents.EventContextV02{
+		ID:         eventID,
+		Type:       cloudEventType,
+		Source:     *a.source,
+		Extensions: extensions,
+	}.AsV02()
+	eventContext.SetSubject(subject)
 
-	// TODO set source and subject properly.
 	event := cloudevents.Event{
-		Context: cloudevents.EventContextV02{
-			ID:         eventID,
-			Type:       cloudEventType,
-			Source:     *source,
-			Extensions: extensions,
-		}.AsV02(),
-		Data: payload,
+		Context: eventContext,
+		Data:    payload,
 	}
-	_, err = a.client.Send(context.TODO(), event)
+	_, err := a.client.Send(context.TODO(), event)
 	return err
 }
 
-func sourceFromGitHubEvent(gitHubEvent gh.Event, payload interface{}) (*types.URLRef, error) {
-	var url string
+func subjectFromGitHubEvent(gitHubEvent gh.Event, payload interface{}) string {
+	// The decision of what to put in subject is somewhat arbitrary here (i.e., it's the author's opinion)
+	// TODO check if we should be setting subject to these values.
+	var subject string
 	switch gitHubEvent {
 	case gh.CheckSuiteEvent:
-		cs := payload.(gh.CheckSuitePayload)
-		url = cs.Repository.HTMLURL
+		if cs, ok := payload.(gh.CheckSuitePayload); ok {
```

Comment:
Do we want to log these failures?

Reply:
I think is not needed in each specific case. 
I just added a generic log at the end of the function to print if the subject == "".

Reply:
Done! Thanks!